Src/VimMac/KeyProcessorProvider.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System;
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio.Commanding;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Editor.Commanding.Commands;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Vim.Mac
+{
+    //[Export(typeof(IKeyProcessorProvider))]
+    //[ContentType("text")]
+    //[TextViewRole(PredefinedTextViewRoles.Interactive)]
+    //[Name(VimConstants.MainKeyProcessorName)]
+    //internal sealed class KeyProcessorProvider2 : IKeyProcessorProvider
+    //{
+    //    private readonly IVim _vim;
+    //    //private readonly IKeyUtil _keyUtil;
+
+    //    [ImportingConstructor]
+    //    internal KeyProcessorProvider2(IVim vim)
+    //    {
+    //        _vim = vim;
+    //    }
+
+    //    public KeyProcessor GetAssociatedProcessor(ICocoaTextView wpfTextView)
+    //    {
+    //        var vimTextBuffer = _vim.GetOrCreateVimBuffer(wpfTextView);
+    //        return new VimKeyProcessor(vimTextBuffer, _keyUtil);
+    //    }
+
+    //}
+
+    /// <summary>
+    /// Passes commands to the IBraceCompletionManager found in the
+    /// property bag of the view.
+    /// </summary>
+    [Export]
+    [Export(typeof(ICommandHandler))]
+    [ContentType(VimConstants.AnyContentType)]
+    [TextViewRole(PredefinedTextViewRoles.Editable)]
+    [Name("Key Processor Provider")]
+    internal class KeyProcessorProvider :
```

Comment:
I couldn't easily use IKeyProcessorProvider so trying out a simpler IChainedCommandHandler<TypeCharCommandArgs>. I don't think this will allow me to check for all key combinations, but it gets me a little further for now.